db/util_testing.go

```diff
@@ -62,18 +82,19 @@ func isPrimaryIndexEmpty(ctx context.Context, store base.N1QLStore) (bool, error
 	// If there was an error, then retry.  Assume it's an "index rollback" error which happens as
 	// the index processes the bucket flush operation
 	if err != nil {
-		return false, err
+		return nil, err
 	}
 
-	// If it's empty, we're done
-	var queryRow map[string]interface{}
-	found := results.Next(ctx, &queryRow)
+	var documents []string
+	var queryRow map[string]string
+	for results.Next(ctx, &queryRow) {
+		documents = append(documents, queryRow["id"])
+	}
 	resultsCloseErr := results.Close()
 	if resultsCloseErr != nil {
-		return false, err
+		return documents, err
```

Comment:
Should this be `resultsCloseErr`? We've already nil checked/returned `err` above.

```suggestion
		return documents, resultsCloseErr
```